nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java

```diff
@@ -0,0 +1,172 @@
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final CEFParser parser = new CEFParser();
+        final byte[] buffer = new byte[(int) flowFile.getSize()];
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                StreamUtils.fillBuffer(in, buffer);
+            }
+        });
+
+        CommonEvent event = null;
+
+        try {
+            event = parser.parse(buffer, true);
+        } catch (Exception e) {
+            // This should never trigger but adding in here as a fencing mechanism to
+            // address possible ParCEFone bugs.
+            getLogger().error("Parser returned unexpected Exception {} while processing {}; routing to failure", new Object[] {e, flowFile});
+            session.transfer(flowFile, REL_FAILURE);
+        }
+
+
+        // ParCEFone returns null every time it cannot parse an
+        // event, so we test
+        if (event==null) {
+            getLogger().error("Failed to parse {} as a CEF message: it does not conform to the CEF standard; routing to failure", new Object[] {flowFile});
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final Map<String, Object> headerFieldMap;
+        final Map<String, Object> extensionFieldMap;
+        try {
+            headerFieldMap = event.getHeader();
+            extensionFieldMap = event.getExtension(true);
+        } catch (CEFHandlingException e) {
+            getLogger().error("Failed to parse {} as a CEF message due to {}; routing to failure", new Object[] {flowFile, e});
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+
+        final Map<String, String> attributes = new HashMap<>();
+
+        // Process KVs of the Header field
+        for (Map.Entry<String, Object> entry : headerFieldMap.entrySet()) {
+            attributes.put("cef.header."+entry.getKey(), prettyResult(entry.getValue()));
+        }
+
+        for (Map.Entry<String, Object> entry : extensionFieldMap.entrySet()) {
+            attributes.put("cef.extension." + entry.getKey(), prettyResult(entry.getValue()));
+        }
+
+        flowFile = session.putAllAttributes(flowFile, attributes);
+        session.transfer(flowFile, REL_SUCCESS);
+    }
+
+    private String prettyResult(Object entryValue) {
+
+        if (entryValue instanceof InetAddress ) {
+            return ((InetAddress) entryValue).getHostAddress();
+        } else if (entryValue instanceof Date) {
+            // Not to discuss. THere should be no date format other than internationally agreed formats...
+            final SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+            // Discuss this
+            // formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
```

Comment:
Like the idea. Something like giving the user the choice to render the output with either GMT+10 or UTC, correct?

Just don't tell me local without a timezone... for that the answer would be over my dead body!!! :skull:  8D